riscv/debug_module.cc

```diff
@@ -844,39 +859,138 @@ bool debug_module_t::perform_abstract_command()
         }
         return true;
 
-      } else {
-        abstractcs.cmderr = CMDERR_NOTSUP;
-        return true;
-      }
-
-      if (is_fpu_reg(regno)) {
-        // restore mstatus
-        write32(debug_abstract, i++, csrr(S0, CSR_DSCRATCH1));
-        write32(debug_abstract, i++, csrw(S0, CSR_MSTATUS));
-        // restore s0
-        write32(debug_abstract, i++, csrr(S0, CSR_DSCRATCH0));
-      }
-    }
-
-    if (get_field(command, AC_ACCESS_REGISTER_POSTEXEC)) {
-      write32(debug_abstract, i,
-          jal(ZERO, debug_progbuf_start - debug_abstract_start - 4 * i));
-      i++;
     } else {
-      write32(debug_abstract, i++, ebreak());
+      abstractcs.cmderr = CMDERR_NOTSUP;
+      return true;
     }
 
-    debug_rom_flags[selected_hart_id()] |= 1 << DEBUG_ROM_FLAG_GO;
-    rti_remaining = config.abstract_rti;
-    abstract_command_completed = false;
+    if (is_fpu_reg(regno)) {
+      // restore mstatus
+      write32(debug_abstract, i++, csrr(S0, CSR_DSCRATCH1));
+      write32(debug_abstract, i++, csrw(S0, CSR_MSTATUS));
+      // restore s0
+      write32(debug_abstract, i++, csrr(S0, CSR_DSCRATCH0));
+    }
+  }
 
-    abstractcs.busy = true;
+  if (get_field(command, AC_ACCESS_REGISTER_POSTEXEC)) {
+    write32(debug_abstract, i,
+        jal(ZERO, debug_progbuf_start - debug_abstract_start - 4 * i));
+    i++;
   } else {
+    write32(debug_abstract, i++, ebreak());
+  }
+
+  debug_rom_flags[selected_hart_id()] |= 1 << DEBUG_ROM_FLAG_GO;
+  rti_remaining = config.abstract_rti;
+  abstract_command_completed = false;
+
+  abstractcs.busy = true;
+  return true;
+}
+
+static unsigned idx(unsigned xlen)
+{
+  return field_width(xlen) - 3U;
+}
+
+bool debug_module_t::perform_abstract_memory_access() {
+  unsigned aamsize = get_field(command, AC_ACCESS_MEMORY_AAMSIZE);
+  bool aampostincrement = get_field(command, AC_ACCESS_MEMORY_AAMPOSTINCREMENT);
+  bool aamvirtual = get_field(command, AC_ACCESS_MEMORY_AAMVIRTUAL);
+  bool is_write = get_field(command, AC_ACCESS_MEMORY_WRITE);
+  auto xlen = sim->get_harts().at(selected_hart_id())->get_xlen();
+
+  if (!selected_hart_state().halted) {
+    abstractcs.cmderr = CMDERR_HALTRESUME;
+    return true;
+  }
+
+  if (aamsize > 3 || aamsize > idx(xlen)) {
```

Comment:
```suggestion
  if (aamsize > idx(xlen)) {
```
Should be enough, no?

Reply:
Addressed